### PR TITLE
chore: Change interface name from Resolver to Router in Configuration

### PR DESCRIPTION
**Pull request by karimalzalek**
Hi ,
Would like some help with rebasing this PR.
This is for (Rename configuration.Resolver to configuration.Router) #1886 

Thanks.

### CODE REVIEW

File: common/configuration/api.go [hunk 1]
```diff
@@ -69,14 +69,14 @@ func (k *Ref) UnmarshalText(text []byte) error {
 	return nil
 }
 
-// A Resolver resolves configuration names to keys that are then used to load
+// A Router (previously named Resolver) resolves configuration names to keys that are then used to load
```
Comment (alecthomas) [inline common/configuration/api.go:72]: No need for this, but thanks :)

File: common/configuration/asm.go [hunk 2]
```diff
@@ -24,18 +24,18 @@ type asmClient interface {
 	delete(ctx context.Context, ref Ref) error
 }
 
-// ASM implements Resolver and Provider for AWS Secrets Manager (ASM).
+// ASM implements Router (previously named Resolver) and Provider for AWS Secrets Manager (ASM).
```
Comment (alecthomas) [inline common/configuration/asm.go:27]: Same here.

File: common/configuration/manager.go [hunk 3]
```diff
@@ -26,10 +26,10 @@ type Configuration struct{}
 func (Configuration) String() string { return "configuration" }
 
 // Manager is a high-level configuration manager that abstracts the details of
-// the Resolver and Provider interfaces.
+// the Router (previously Resolver) and Provider interfaces.
```
Comment (alecthomas) [inline common/configuration/manager.go:29]: Same here.

**Review by alecthomas — APPROVED** [review]
Appreciate it, thanks for the PR! Just a couple of minor comments.

**Comment by karimalzalek** [comment]
Hello @alecthomas , removed the comments 👍

**Comment by karimalzalek** [comment]
No problem @alecthomas , I noticed some failed tests, do you want to look at these ?

**Comment by alecthomas** [comment]
They're not test failures, they're automation failures due to token permission issues because you're a new contributor.

**Comment by karimalzalek** [comment]
Ah just noticed it seems to be permissions on remote repos that I don't have access to, all good then 👍

**Pull request merged**